core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -107,9 +107,7 @@ private <T extends Identity> SdkHttpFullRequest doSraSign(SdkHttpFullRequest req
 
     private SdkHttpFullRequest toSdkHttpFullRequest(SignedRequest signedRequest) {
         SdkHttpRequest request = signedRequest.request();
-        if (request instanceof SdkHttpFullRequest) {
-            return (SdkHttpFullRequest) request;
-        }
```

Comment:
Is this old code and what's the story behind the removal?

Reply:
Yes, sorry for not clarifying. This code is problematic, because in cases where the request IS an SdkHttpFullRequest and the payload is transformed in signing, the underlying payload never gets re-set to the transformed payload from signing.